modules/apps/app-builder/app-builder-test/src/testIntegration/java/com/liferay/app/builder/service/test/ProductMenuAppDeployerTest.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.app.builder.service.test;
```

Comment:
The package has to follow the same package for product app menu

Reply:
Actually , i think we can make this test a little bit more generic and name it AppDeployerTest so we can put other types of tests(standalone, layout) there later, what you think?  If it makes sense the package should be the same as the AppDeployer package in app-builder-api

Reply:
There is an `AppBuilderAppLocalServiceTest` test already that covers something regarding _standaalone_ deployments. Should it be reused?

Reply:
Don't think so. The validations is there because of this: https://github.com/liferay/liferay-portal/blob/5c10ae66910f788846cb111cb401b85cbd3a4e43/modules/apps/app-builder/app-builder-service/src/main/java/com/liferay/app/builder/service/impl/AppBuilderAppDeploymentLocalServiceImpl.java#L71

When we delete an AppBuilderAppDeployment we are undeploying the deployed apps. It's there just to make sure that an exception doesn't happen if we have a deployed app

Reply:
Hi Jey! About the package name i should create a new package called `com.liferay.app.builder.deploy.test` then? It's the one for AppDeployer

I thought before that the package should be `com.liferay.app.builder.service.test` since the test is being made at the same folder as the other test AppBuilderAppLocalServiceTest

Reply:
Hi @rebsilva . Yeah, you should see [this test class](https://github.com/liferay/liferay-portal/blob/master/modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/data/provider/test/DDMDataProviderTest.java) as example. It's also testing something that is exposed on the API .